src/generic/UnivPoly.jl

```diff
@@ -24,7 +24,7 @@ function is_exact_type(::Type{T}) where {S <: RingElement, U <: MPolyRingElem{S}
    return is_exact_type(S)
 end
 
-parent(p::UnivPoly) = p.parent
+parent(p::UnivPoly) = p.parent::parent_type(p)
```

Comment:
somehow julia couldn't infer `p.parent` properly, the second type param (the MPolyRing) abstract. Additionally use this accessor everywhere in this file instead of direct access